### PR TITLE
[WIP] Image to HEIF tool

### PR DESCRIPTION
**Pull request by obilaniu**
This PR exists to track progress on the `image2heif` conversion tool.

### CODE REVIEW

**Comment by obilaniu** [comment]
@satyaog This PR is far from done though. I need to finish the middle-end and move on to back-end matters.

**Comment by satyaog** [comment]
@obilaniu , ok I believe I'll receive emails notifying the next commits as they come

File: src/tools/image2heif/main.c [hunk 1]
```diff
@@ -1786,30 +1812,33 @@ static int  i2h_universe_item_handle_picture      (UNIVERSE* u, ITEM* item){
...
+     *    5. Release encoder.
+     *    6. Split/duplicate single list of packets between thumbnail and grid tiles.
+     *    7. Create new items and fill them.
+     *    8. If this item is not single-tile,
+     *        8.1 Transmute to "grid".
+     *        8.2 Compute grid references.
      */
     
-    /**
-     * Push encoded tiles into the encoder.
```
Comment (fosterrath-mila) [inline src/tools/image2heif/main.c]: Might want to hold on to the comments no ?

Comment (obilaniu) [reply src/tools/image2heif/main.c]: The way the middle-end is written, with descriptive function names and error messages, it's somewhat redundant.

**Comment by fosterrath-mila** [comment]
Thanks Olexa!

**Comment by fosterrath-mila** [comment]
All good

**Comment by fosterrath-mila** [comment]
Could we do without the benzina.h include ? (I'm thinking of GPL implications)

https://github.com/obilaniu/Benzina/blob/ca54ed89572d502d48e9d032e29c6f974420fac5/src/tools/image2heif/main.c#L23

**Comment by fosterrath-mila** [comment]
Also, would be good to document somewhere the system dependencies for running benzina. 

e.g. libavcodec, libavutil, libavfilter ... any more ?

**Comment by obilaniu** [comment]
@fosterrath-mila Re GPL:

- `libbenzina.so` is a standalone library licensed MIT.
- `libbenzina.so` does not derive from `libx265.so` or from any GPL'ed code.
- `image2heif` is a program using `libbenzina.so` and a GPL'ed build of FFmpeg via `libx265.so`.
- `image2heif` is therefore licensed GPL. The work as a whole (`image2heif`, `libbenzina.so`, `libavcodec.so`, `libx265.so`) is licensed GPL (because of the viral nature of the GPL), but the individual parts remain licensed under their respective licenses.

Take for instance X.org, which is MIT-licensed, but many GPL programs use it. It's not thereby made GPL itself.

To achieve clear separation and as a safeguard,
- The Meson build definitions by default set [`ENABLE_GPL=false`](https://github.com/obilaniu/Benzina/blob/master/meson_options.txt#L6),
- The `setup.py` file only enables GPL [if the environment variable `ENABLE_GPL` is true](https://github.com/obilaniu/Benzina/blob/master/scripts/utils.py#L55) and
- The building of `image2heif` is [guarded with a license check `ENABLE_GPL=true`](https://github.com/obilaniu/Benzina/blob/master/src/tools/image2heif/meson.build#L9).

**Comment by obilaniu** [comment]
@fosterrath-mila I wouldn't be opposed to making the separation even more obvious by splitting out things into a `gpl/` subdirectory.

**Comment by obilaniu** [comment]
@fosterrath-mila Re other dependencies: The [`meson.build`](https://github.com/obilaniu/Benzina/blob/master/meson.build) is quite readable. The only other place where external `dependency()` or `find_library()` are accessed is under `thirdparty/`, which concerns CUDA.

I build a stub `libnvcuvid.so.1` myself, so I don't need to depend on its presence on any platform.

**Comment by obilaniu** [comment]
@satyaog @fosterrath-mila I'm sorry about the force-pushes, but I screwed up and somehow lost your merge commits  on `master`. I just re-merged your changes back into `master` and this time properly rebased this branch (`image2heif`) atop `master`. That's now done, but your copies are almost certainly stale now.

**Pull request merged**